diff --git a/arch/arm/boot/dts/amlogic/meson6_g18.dtd b/arch/arm/boot/dts/amlogic/meson6_g18.dtd
index 962cec6..36a56ed 100755
--- a/arch/arm/boot/dts/amlogic/meson6_g18.dtd
+++ b/arch/arm/boot/dts/amlogic/meson6_g18.dtd
@@ -756,6 +756,7 @@ void root_func(){
 
 /// ***************************************************************************************
 ///	-	Nand
+//$$ MODULE="Nand"
 //$$ DEVICE="aml_nand"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2 ="reg"
@@ -765,7 +766,7 @@ void root_func(){
 //$$ L3 PROP_STR 2 ="plat-names"
 //$$ L2 PROP_U32 = "plat-num"
 //$$ L2 PROP_CHOICE "plat-part-0_match" = "plat-part-0"
-//$$ L2 PROP_CHOICE "plat-part-1_match" = "plat-part-1"
+// L2 PROP_CHOICE "plat-part-1_match" = "plat-part-1"
 	nand{
 		compatible = "amlogic,aml_nand";
 		dev_name = "nand";
@@ -775,38 +776,15 @@ void root_func(){
 		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base
 									&nand_ce0 &nand_rb0
 									&nand_rb1 &nand_ce1>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base 
 									&nand_ce0 &nand_ce1
 									&nand_ce2 &nand_ce3>;
 		device_id = <0>;
-		plat-names = "bootloader","nandnormal";
-		plat-num = <2>;
-		plat-part-0 = <&bootload>;
-		plat-part-1 = <&normal>;
-
-//$$ MATCH "plat-part-0_match" = <&bootload>
-//$$ L2 PROP_STR = "enable_pad"
-//$$ L2 PROP_STR = "busy_pad"
-//$$ L2 PROP_STR = "timming_mode"
-//$$ L2 PROP_STR = "bch_mode"
-//$$ L2 PROP_U32 = "t_rea"
-//$$ L2 PROP_U32 = "t_rhoh"
-//$$ L2 PROP_U32 = "chip_num"
-//$$ L2 PROP_U32 = "part_num"
-//$$ L2 PROP_U32 = "rb_detect"
-		bootload: bootload{
-			enable_pad ="ce0";
-			busy_pad = "rb0";
-			timming_mode = "mode5";
-			bch_mode = "bch60_1k";
-			t_rea = <20>;
-			t_rhoh = <15>;
-			chip_num = <1>;
-			part_num = <0>;
-			rb_detect = <1>;
-		};
+		plat-names = "nandnormal";
+		plat-num = <1>;
+		plat-part-0 = <&normal>;
 
-//$$ MATCH "plat-part-1_match" = <&normal>
+//$$ MATCH "plat-part-0_match" = <&normal>
 //$$ L2 PROP_STR 2 = "enable_pad"
 //$$ L2 PROP_STR 2 = "busy_pad"
 //$$ L2 PROP_STR = "timming_mode"
@@ -826,41 +804,33 @@ void root_func(){
 			t_rea = <20>;
 			t_rhoh = <15>;
 			chip_num = <2>;
-			part_num = <7>;
+			part_num = <8>;
 			partition = <&nand_partitions>;
 			rb_detect = <1>;
 		};
 
 		nand_partitions:nand_partition{
 			logo{
-				offset=<0x0 0x4800000>;
-				size=<0x0 0x800000>;
-			};
-			aml_logo{
-				offset=<0x0 0x5800000>;
-				size=<0x0 0x800000>;
+				offset=<0x0 0x0>;
+				size=<0x0 0x400000>;
 			};
 			recovery{
-				offset=<0x0 0x6800000>;
+				offset=<0x0 0x400000>;
 				size=<0x0 0x800000>;
 			};
 			boot{
-				offset=<0x0 0x8800000>;
+				offset=<0x0 0xC00000>;
 				size=<0x0 0x800000>;
 			};
 			system{
-				offset=<0x0 0xa800000>;
-				size=<0x0 0x40000000>;
+				offset=<0x0 0x1400000>;
+				size=<0x0 0x10000000>;
 			};
 			cache{
-				offset=<0x0 0x4a800000>;
-				size=<0x0 0x20000000>;
-			};
-			backup{
-				offset=<0x0 0x6a800000>;
+				offset=<0x0 0x11400000>;
 				size=<0x0 0x10000000>;
 			};
-			userdata{
+			data{
 				offset=<0xffffffff 0xffffffff>;
 				size=<0x0 0x0>;
 			};
