From a44a4cda9c1cc8392e389a59590a1183cb611454 Mon Sep 17 00:00:00 2001
From: Quihico <Quihico@gmail.com>
Date: Wed, 25 Nov 2015 17:09:09 +0100
Subject: [PATCH 05/16] Added Connectivity_Check-function

---
 src/oe.py | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/src/oe.py b/src/oe.py
index 7b7c50b..3b6eb33 100644
--- a/src/oe.py
+++ b/src/oe.py
@@ -485,6 +485,35 @@ def openWizard():
         xbmc.executebuiltin('Dialog.Close(busydialog)')
         dbg_log('oe::openWizard', 'ERROR: (' + repr(e) + ')')
 
+def Connectivity_Check():
+    internetcheck = 1
+    try:
+        Open_URL('http://google.com')
+    except:
+        try:
+            Open_URL('http://google.com')
+        except:
+            try:
+                Open_URL('http://google.com')
+            except:
+                try:
+                    Open_URL('http://google.cn')
+                except:
+                    try:
+                        Open_URL('http://google.cn')
+                    except:
+                        internetcheck=0
+    if internetcheck==1:
+        xbmc.executebuiltin('RunAddon(script.openwindow)')
+    if internetcheck==0:
+        datapath = xbmc.translatePath(os.path.join('special://profile/addon_data/service.openelec.settings/',''))
+        if os.path.exists(datapath)==True:
+            shutil.rmtree(datapath)
+        try:
+            xbmc.executebuiltin('RunScript(/usr/share/kodi/addons/service.openelec.settings/service.py)')
+        except:
+            xbmc.executebuiltin('RunScript(special://home/addons/service.openelec.settings/service.py)')
+
 
 def Open_URL(url):
     req = urllib2.Request(url)
-- 
1.9.1

