From 769186a1bfb2afb76c0b134783a3239fe4eca10a Mon Sep 17 00:00:00 2001
From: Quihico <Quihico@gmail.com>
Date: Fri, 27 Nov 2015 09:15:19 +0100
Subject: [PATCH 10/16] Added branding plan b

---
 src/oe.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/oe.py b/src/oe.py
index c1838fb..65668fe 100644
--- a/src/oe.py
+++ b/src/oe.py
@@ -801,11 +801,18 @@ else:
     SYSTEMID = os.environ.get('SYSTEMID', '')
 
 ############################################################################################
+branding = 'media/BRANDING/BRANDING.zip'
 
 try:
     configFile = '%s/userdata/addon_data/service.openelec.settings/oe_settings.xml' % XBMC_USER_HOME
     if not os.path.exists('%s/userdata/addon_data/service.openelec.settings' % XBMC_USER_HOME):
         os.makedirs('%s/userdata/addon_data/service.openelec.settings' % XBMC_USER_HOME)
+    if os.path.exists(branding):
+        try:
+            extract.all(branding,'/storage')
+            dialog.ok('SUCCESS','OpenELEC Branding successfully copied to device.')
+        except:
+            dialog.ok('FAILED!!!','Failed to copy OpenELEC Branding to device!!!')
     if not os.path.exists('%s/services' % CONFIG_CACHE):
         os.makedirs('%s/services' % CONFIG_CACHE)
 except:
-- 
1.9.1

